src/v2/blobs.rs

```diff
@@ -99,6 +164,10 @@ impl Client {
                         status
                     )))
                 }
+            }).
+            and_then(|body| {
+                layer_digest?.try_verify(&body)?;
```

Comment:
Perhaps we should move `layer_digest?` earlier (or as a different input type for the function) and avoid hitting the network if we already know the digest is invalid.

Reply:
I moved it into a separate future and serialized the two.